### PR TITLE
Allow overriding the duration of the job type in the recurring job

### PR DESCRIPTION
**Pull request by pascalfree**
Not urgent

I got into the situation a few times where the only reason I couldn't use a jobtype, was that it's duration was not matching.
Particular examples are after-work jobs in the greenhouse that get shorter at the end of the season as the days get shorter and IT meetings that may start at a different time, while the end time might stay the same.
This can be resolved in multiple ways that all have their disatvantages:

1. Change the duration in the type: This is not feasible if there are other jobs of this type in the future that have another duration. Even if from now on the job will have the new different duration, this would change the duration of all past jobs of that type wich is particularly unsuitable if the member assignments are counted in hours.
2. Keep the duration anyways: This will raise a wrong expectation or might confuse members.
3. Create a new identical job type with the new duration. This will clutter the job types and make it hard to choose the job type in the dropdown, as the duration is not visible there. (sidenote: probably it would be better to show the unique name instead of the display name in that dropdown) Also the description of the job type will likely get fragmented.

The solution suggested here allow to set the duration per recurring job instance, by optionally overriding the duration set in the job type.

Sidenode: If assignments are counted by hours, changing the job type duration should be prohibited or at least give a warning. With this pull request in place it would be technically easy to automatically set the duration of past jobs to the old duration of the job type if the latter is changed (except for the past jobs that already had an overridden duration).

I implemented this to see how many changes would actually be needed (not so many in my view), so now it's a pull request instead of an issue.
I renamed the field on the job type to give the nicer name to the property (which would conflict on the OneTimeJob).

### CODE REVIEW

File: juntagrico/entity/jobs.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ class AbstractJobType(JuntagricoBaseModel):
     displayed_name = models.CharField(_('Angezeigter Name'), max_length=100, blank=True, null=True)
     description = models.TextField(_('Beschreibung'), max_length=1000, default='')
     activityarea = models.ForeignKey(ActivityArea, on_delete=models.PROTECT, verbose_name=_('Tätigkeitsbereich'))
-    duration = models.PositiveIntegerField(_('Dauer in Stunden'))
+    default_duration = models.PositiveIntegerField(_('Dauer in Stunden'))
```
Comment (motu55) [inline juntagrico/entity/jobs.py:108]: ad a help text or so that it is clear that this value is a default value

File: juntagrico/entity/jobs.py [hunk 2]
```diff
@@ -233,6 +237,14 @@ class Meta:
 class RecuringJob(Job):
     type = models.ForeignKey(JobType, on_delete=models.PROTECT, verbose_name=_('Jobart'))
     additional_description = models.TextField(_('Zusätzliche Beschreibung'), max_length=1000, blank=True, default='')
+    duration_override = models.PositiveIntegerField(
+        _('Besondere Dauer in Stunden'), null=True, blank=True, default=None,
```
Comment (motu55) [inline juntagrico/entity/jobs.py:241]: i would not call it besondere, somethin more hinti that is is an override

File: juntagrico/migrations/0021_duration_override.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
+# Generated by Django 3.0.3 on 2020-03-29 19:22
```
Comment (motu55) [inline juntagrico/migrations/0021_duration_override.py:1]: wieso ist die so gross?

Comment (pascalfree) [reply juntagrico/migrations/0021_duration_override.py:1]: Django macht eine migration für die help texts und die migration wurde beim letzten mal nicht gemacht.

File: juntagrico/entity/jobs.py (hunk 2 above)
Comment (pascalfree) [reply juntagrico/entity/jobs.py]: tried my best

**Comment by motu55** [comment]
will merge once #204 is merged due to the migrations part
also remove the migrations here we will do them again in the master

**Pull request merged**